### ISSUE TITLE
RuntimeError: detect_exception_in_thread_pool.load()

### ISSUE THREAD

**Issue by wprazuch** [Lightning-AI/lightning-thunder#671]
## 🐛 Bug

There is a CUDA error running for:
- falcon-7b
single/multi-gpu (ddp and fsdp zero2) with Thunder compile mode

### To Reproduce

(example for one model)

```
mkdir -p output
docker run --pull=always --gpus all --ipc=host --ulimit memlock=-1 --ulimit stack=67108864  -v $PWD/output:/output -it INTERNAL_IMAGE:pjnl-20240621
```
Run in the container:

```
torchrun --nproc-per-node=8 /opt/pytorch/lightning-thunder/thunder/benchmarks/benchmark_litgpt.py --model_name falcon-7b --compile thunder_cudnn --distributed_mode fsdp --shard_mode zero2 
```

### Expected behavior

The model should run or we should get OOM error.

### Environment
As in Docker image.

### Additional context

Traceback:
```
[rank0]: Traceback (most recent call last):
[rank0]:   File "/workspace/lightning-thunder/thunder/benchmarks/benchmark_litgpt.py", line 584, in benchmark_main
[rank0]:     benchmark.train()
[rank0]:   File "/workspace/lightning-thunder/thunder/benchmarks/benchmark_litgpt.py", line 485, in train
[rank0]:     logits = self.model(input_ids)
[rank0]:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1657, in _wrapped_call_impl
[rank0]:     return self._call_impl(*args, **kwargs)
[rank0]:   File "/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py", line 1675, in _call_impl
[rank0]:     return forward_call(*args, **kwargs)
[rank0]:   File "/opt/pytorch/lightning-thunder/thunder/core/module.py", line 61, in forward
[rank0]:     res = self._forward_fn(*args, **kwargs)
[rank0]:   File "/opt/pytorch/lightning-thunder/thunder/__init__.py", line 674, in fn_
[rank0]:     result = cache_entry.computation_fn(*inps)
[rank0]:   File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 116, in decorate_context
[rank0]:     return func(*args, **kwargs)
[rank0]:   File "/usr/local/lib/python3.10/dist-packages/torch/amp/autocast_mode.py", line 43, in decorate_autocast
[rank0]:     return func(*args, **kwargs)
[rank0]:   File "/usr/local/lib/python3.10/dist-packages/torch/amp/autocast_mode.py", line 43, in decorate_autocast
[rank0]:     return func(*args, **kwargs)
[rank0]:   File "thunder.augmented_forward_fn_2", line 721, in augmented_forward_fn
[rank0]:   File "/opt/pytorch/lightning-thunder/thunder/executors/nvfuserex_impl.py", line 412, in __call__
[rank0]:     return fd.execute(args, **kwargs)
[rank0]:   File "/opt/pytorch/nvfuser/nvfuser/__init__.py", line 145, in execute
[rank0]:     result = self._execute(
[rank0]: RuntimeError: !detect_exception_in_thread_pool.load() INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/kernel_cache.cpp":1336, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Detected exception while compiling fusion segments in parallel. Error messages from all threads are printed below.
```

**Comment by wprazuch** [comment]
Sorry, closing as duplicate of: https://github.com/Lightning-AI/lightning-thunder/issues/583
